src/Android/Xamarin.Android/Samples/Utility network/DisplayUtilityAssociations/DisplayUtilityAssociations.cs

```diff
@@ -0,0 +1,183 @@
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            try
+            {
+                // Create the utility network.
+                _utilityNetwork = await UtilityNetwork.CreateAsync(new Uri(FeatureServerUrl));
+
+                _myMapView.Map = new Map(Basemap.CreateTopographicVector());
+
+                // Get all of the edges and junctions in the network.
+                IEnumerable<UtilityNetworkSource> edges = _utilityNetwork.Definition.NetworkSources.Where(n => n.SourceType == UtilityNetworkSourceType.Edge);
+                IEnumerable<UtilityNetworkSource> junctions = _utilityNetwork.Definition.NetworkSources.Where(n => n.SourceType == UtilityNetworkSourceType.Junction);
+
+                // Add all edges that are not subnet lines to the map.
+                foreach (UtilityNetworkSource source in edges)
+                {
+                    if (source.SourceUsageType != UtilityNetworkSourceUsageType.SubnetLine && source.FeatureTable != null)
+                    {
+                        _myMapView.Map.OperationalLayers.Add(new FeatureLayer(source.FeatureTable));
+                    }
+                }
+
+                // Add all junctions to the map.
+                foreach (UtilityNetworkSource source in junctions)
+                {
+                    if (source.FeatureTable != null)
+                    {
+                        _myMapView.Map.OperationalLayers.Add(new FeatureLayer(source.FeatureTable));
+                    }
+                }
+
+                // Create a graphics overlay for associations.
+                _associationsOverlay = new GraphicsOverlay();
+                _myMapView.GraphicsOverlays.Add(_associationsOverlay);
+
+                // Populate the legend in the UI.
+                RuntimeImage attachmentSwatch = await _attachmentSymbol.CreateSwatchAsync();
+                Android.Graphics.Bitmap attachmentBitmap = await attachmentSwatch?.ToImageSourceAsync();
+                _attachmentImageView.SetImageBitmap(attachmentBitmap);
+                RuntimeImage connectSwatch = await _connectivitySymbol.CreateSwatchAsync();
+                Android.Graphics.Bitmap connectivityBitmap = await connectSwatch?.ToImageSourceAsync();
+                _connectivityImageView.SetImageBitmap(connectivityBitmap);
+
+                // Set the starting viewpoint.
+                await _myMapView.SetViewpointAsync(InitialViewpoint);
+            }
+            catch (Exception ex)
+            {
+                new AlertDialog.Builder(this).SetMessage(ex.Message).SetTitle(ex.GetType().Name).Show();
+            }
+        }
+
+        private async void OnNavigationCompleted(object sender, EventArgs e)
+        {
+            try
+            {
+                // Check if the current viewpoint is outside of the max scale.
+                if (_myMapView.GetCurrentViewpoint(ViewpointType.CenterAndScale)?.TargetScale >= _maxScale)
```

Comment:
Outside of that scale value all of the associations in this network are not visible, and the calls to get them are more costly and can produce the toomanyassociations error often. Doing this instead of setting the map value keeps the ability to view the whole network, even if its not useful/feasible to render associations at that scale.